### PR TITLE
work for today

### PR DESCRIPTION
**Pull request by frog404**
in NO3, matched this 7 functions:

- func_801BE9F4
- func_801C4F64
- func_801C4FA0
- func_801C55E8
- func_801C5620
- func_801C5794
- func_801C57C4

### CODE REVIEW

File: src/st/no3/377D4.c [hunk 1]
```diff
@@ -412,9 +419,27 @@ INCLUDE_ASM("asm/st/no3/nonmatchings/377D4", func_801C4D4C);
...
 INCLUDE_ASM("asm/st/no3/nonmatchings/377D4", func_801C4E4C);
 
-INCLUDE_ASM("asm/st/no3/nonmatchings/377D4", func_801C4F64);
+s16 func_801C4F64(void) {
+    s16 temp_v1;
+    s16 var_a0;
+
+    temp_v1 = (u16) D_8006C3B8->posX.Data.high - (u16) D_800733DA;
+    var_a0 = temp_v1;
+    if (temp_v1 >> 0x10) {
```
Comment (pixel-stuck) [inline src/st/no3/377D4.c:428]: This can probably be written as `if (temp_v1 < 0)`

File: src/st/no3/377D4.c [hunk 2]
```diff
@@ -463,19 +488,31 @@ s16 func_801C5560(s32 arg0) { return D_801820C4[arg0 & 0xFF]; }
 
 INCLUDE_ASM("asm/st/no3/nonmatchings/377D4", func_801C557C);
 
-INCLUDE_ASM("asm/st/no3/nonmatchings/377D4", func_801C55E8);
+s32 func_801C55E8(s16 arg0, s16 arg1) {
```
Comment (pixel-stuck) [inline src/st/no3/377D4.c:491]: This function probably actually returns u8, and the & 0xFF at the end can be gotten rid of

File: src/st/no3/377D4.c [hunk 3]
```diff
@@ -463,19 +488,31 @@ s16 func_801C5560(s32 arg0) { return D_801820C4[arg0 & 0xFF]; }
...
+    return ((ratan2((s32) arg1, (s32) arg0) >> 4) + 0x40) & 0xFF;
+}
 
-INCLUDE_ASM("asm/st/no3/nonmatchings/377D4", func_801C5620);
+s32 func_801C5620(Entity *arg0, Entity *arg1) {
```
Comment (pixel-stuck) [inline src/st/no3/377D4.c:495]: Similar here, returns u8

File: src/st/no3/377D4.c [hunk 4]
```diff
@@ -463,19 +488,31 @@ s16 func_801C5560(s32 arg0) { return D_801820C4[arg0 & 0xFF]; }
...
 
 INCLUDE_ASM("asm/st/no3/nonmatchings/377D4", func_801C5668);
 
 INCLUDE_ASM("asm/st/no3/nonmatchings/377D4", func_801C56B0);
 
 INCLUDE_ASM("asm/st/no3/nonmatchings/377D4", func_801C5708);
 
-INCLUDE_ASM("asm/st/no3/nonmatchings/377D4", func_801C5794);
+u16 func_801C5794(s16 arg0, s16 arg1) {
+    return ratan2((s32) arg1, (s32) arg0);
```
Comment (pixel-stuck) [inline src/st/no3/377D4.c:508]: The casts on these probably aren't required

File: src/st/no3/377D4.c [hunk 5]
```diff
@@ -17,6 +17,9 @@ extern ObjectInit* D_801D7110;
 extern ObjectInit* D_801D7114;
 extern s8 D_801D7118;
 extern s8 D_801D711C;
+extern s16 D_801D7D60;
+extern s16 D_801D7D62;
+extern s32 D_801D7D64;
 
 void SpawnExplosionEntity(u16, Entity*);
 void ReplaceCandleWithDrop(Entity*);
```
Comment (sonicdcer) [inline src/st/no3/377D4.c:25]:
```suggestion
void SpawnExplosionEntity(u16, Entity*);
void ReplaceCandleWithDrop(Entity*);
void EntityCandleDrop(Entity*);
void EntityCandleHeartDrop(Entity*);

extern ObjectInit* D_801D7114;
extern s8 D_801D7118;
extern s8 D_801D711C;
extern s16 D_801D7D60;
extern s16 D_801D7D62;
extern s32 D_801D7D64;
```
can you put the function prototypes at the top of the file please?

File: src/st/no3/377D4.c [hunk 6]
```diff
@@ -412,9 +419,27 @@ INCLUDE_ASM("asm/st/no3/nonmatchings/377D4", func_801C4D4C);
...
+    s16 temp_v1;
+    s16 var_a0;
+
+    temp_v1 = (u16) D_8006C3B8->posX.Data.high - (u16) D_800733DA;
+    var_a0 = temp_v1;
+    if (temp_v1 >> 0x10) {
+        var_a0 = -temp_v1;
+    }
+    return var_a0;
+}
```
Comment (sonicdcer) [inline src/st/no3/377D4.c:432]:
```suggestion
    s16 temp_v1;

    temp_v1 = (u16) D_8006C3B8->posX.Data.high - (u16) D_800733DA;
    var_a0 = temp_v1;
    if (temp_v1 >> 0x10) {
        temp_v1 = -temp_v1;
    }
    return temp_v1;
}
```
The temp might not be necessary, does this match?

**Review by sonicdcer — COMMENTED** [review]
Great work! just a couple of things

File: src/st/no3/377D4.c (hunk 6 above)
Comment (frog404) [reply src/st/no3/377D4.c:432]: one variable was unecessary indeed !
this matched:

s16 func_801C4F64(void) {
    s16 temp_v1;

    temp_v1 = D_8006C3B8->posX.Data.high - D_800733DA;
    if (temp_v1 < 0) {
        temp_v1 = -temp_v1;
    }
    return temp_v1;
}

**Review by frog404 — COMMENTED** [review]
implementing fixes as per sonicdcer and pixel-stuck review

**Review by Xeeynamo — APPROVED** [review]

**Pull request merged**